convert2rhel/unit_tests/actions/is_loaded_kernel_latest_test.py

```diff
@@ -0,0 +1,579 @@
+        )
+
+        is_loaded_kernel_latest_action.run()
+
+        assert message in caplog.records[-1].message
+
+    @pytest.mark.parametrize(
+        (
+            "repoquery_version",
+            "uname_version",
+            "return_code",
+            "package_name",
+        ),
+        (
+            (
+                "C2R\t1634146676\t3.10.0-1160.45.1.el7\tbaseos",
+                "3.10.0-1160.42.2.el7.x86_64",
+                0,
+                "kernel-core",
+            ),
+        ),
+    )
+    @centos8
+    def test_is_loaded_kernel_latest_eus_system_invalid_kernel_version(
+        self,
+        pretend_os,
+        repoquery_version,
+        uname_version,
+        return_code,
+        package_name,
+        tmpdir,
+        monkeypatch,
+        caplog,
+        is_loaded_kernel_latest_action,
+    ):
+        fake_reposdir_path = str(tmpdir)
+        monkeypatch.setattr(
+            actions.is_loaded_kernel_latest,
+            "get_hardcoded_repofiles_dir",
+            value=lambda: fake_reposdir_path,
+        )
+
+        monkeypatch.setattr(actions.is_loaded_kernel_latest.system_info, "has_internet_access", True)
+
+        run_subprocess_mocked = mock.Mock(
+            spec=run_subprocess,
+            side_effect=run_subprocess_side_effect(
+                (
+                    (
+                        "repoquery",
+                        "--setopt=exclude=",
+                        "--quiet",
+                        "--qf",
+                        "C2R\\t%{BUILDTIME}\\t%{VERSION}-%{RELEASE}\\t%{REPOID}",
+                        "--setopt=reposdir=%s" % fake_reposdir_path,
+                        package_name,
+                    ),
+                    (
+                        repoquery_version,
+                        return_code,
+                    ),
+                ),
+                (("uname", "-r"), (uname_version, return_code)),
+            ),
+        )
+        monkeypatch.setattr(
+            actions.is_loaded_kernel_latest,
+            "run_subprocess",
+            value=run_subprocess_mocked,
+        )
+
+        is_loaded_kernel_latest_action.run()
+
+        repoquery_kernel_version = repoquery_version.split("\t")[2]
+        uname_kernel_version = uname_version.rsplit(".", 1)[0]
+        assert is_loaded_kernel_latest_action.error_id == "INVALID_KERNEL_VERSION"
+        assert is_loaded_kernel_latest_action.status == actions.STATUS_CODE["ERROR"]
+        assert (
+            "The version of the loaded kernel is different from the latest version in repositories defined in the %s folder"
+            % fake_reposdir_path
+            in is_loaded_kernel_latest_action.message
+        )
+        assert (
+            "Latest kernel version available in baseos: %s\n" % repoquery_kernel_version
+            in is_loaded_kernel_latest_action.message
+        )
+        assert "Loaded kernel version: %s\n" % uname_kernel_version in is_loaded_kernel_latest_action.message
```

Comment:
I think it would be best to try and use the `assert_actions_result` here. 

Maybe you could try to reconstruct the message to use with that method. If that does not work, then I'm fine not having it.